hist/hist/src/TGraph.cxx

```diff
@@ -2008,6 +2011,62 @@ Int_t TGraph::RemovePoint(Int_t ipoint)
    return ipoint;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Save the graph as .csv, .tsv or .txt.
+///
+/// The result can be immediately imported into Excel, gnuplot, Python or whatever,
+/// without the needing to install pyroot, etc.
+///
+/// the file extension defines the delimiter used:
+///  - `.csv` : comma
+///  - `.tsv` : tab
+///  - `.txt` : space
+
+void TGraph::SaveAs(const char *filename, Option_t *option) const
+{
+   TString del = "";
+   if (filename) {
+      if (strstr(filename,".csv")) del = ",";
+      if (strstr(filename,".tsv")) del = "\t";
+      if (strstr(filename,".txt")) del = " ";
+   }
+   if (del.Length()>0) {
+      std::ofstream out;
+      out.open(filename, std::ios::out);
+      if (!out.good ()) {
+         Error("SaveAs", "cannot open file: %s", filename);
+         return;
+      }
+      if (this->InheritsFrom(TGraphErrors::Class()) ) {
+         if(strstr(option,"title"))
+         out << "# " << GetXaxis()->GetTitle() << "\tex\t" << GetYaxis()->GetTitle() << "\tey" << std::endl;
+         double *ex = this->GetEX();
+         double *ey = this->GetEY();
+         for(int i=0 ; i<fNpoints ; i++)
+         out << fX[i] << del.Data() << ex[i] << del.Data() << fY[i] << del.Data() << ey[i] << std::endl;
+      } else if (this->InheritsFrom(TGraphAsymmErrors::Class()) || this->InheritsFrom(TGraphBentErrors::Class())) {
+         if(strstr(option,"title"))
+         out << "# " << GetXaxis()->GetTitle() << "\texl\t" << "\texh\t" << GetYaxis()->GetTitle() << "\teyl" << "\teyh" << std::endl;
+         double *exl = this->GetEXlow();
+         double *exh = this->GetEXhigh();
+         double *eyl = this->GetEYlow();
+         double *eyh = this->GetEYhigh();
+         for(int i=0 ; i<fNpoints ; i++)
+         out << fX[i] << del.Data() << exl[i] << del.Data() << exh[i] << del.Data() << fY[i] << del.Data() << eyl[i] << del.Data() << eyh[i] << std::endl;
+      } else {
+         if (strstr(option,"title"))
+         out << "# " << GetXaxis()->GetTitle() << "\t" << GetYaxis()->GetTitle() << std::endl;
+         for (int i=0 ; i<fNpoints ; i++)
+         out << fX[i] << del.Data()  << fY[i] << std::endl;
+      }
+      out.close();
+      Info("SaveAs", "csv file: %s has been generated", filename);
```

Comment:
```suggestion
      Info("SaveAs", "%s file: %s has been generated", del.Data(), filename);
```

Reply:
Good point but that does not work. We need another TString (see my commit).

Reply:
o yeah, you are right. Thanks!